components/scream/src/physics/shoc/atmosphere_macrophysics.cpp

```diff
@@ -22,30 +27,85 @@ void SHOCMacrophysics::set_grids(const std::shared_ptr<const GridsManager> grids
+  //       using the same approach to make it easier to follow.
 
-  auto VL = FieldTag::VerticalLevel;
-  auto CO = FieldTag::Column;
+  // These variables are needed by the interface, but not actually passed to shoc_main.
+  m_required_fields.emplace("pref_mid",scalar2d_layout_lev, Pa, grid_name);
+  m_required_fields.emplace("t",       scalar3d_layout_mid, nondim, grid_name);
+  m_required_fields.emplace("alst",    scalar3d_layout_mid, Pa,     grid_name);
+  m_required_fields.emplace("zi",      scalar3d_layout_int, m,      grid_name);
+  m_required_fields.emplace("zm",      scalar3d_layout_mid, K,      grid_name);
+  m_required_fields.emplace("omega",   scalar3d_layout_mid, K,      grid_name);
+  m_required_fields.emplace("shf",     scalar2d_layout_col, K,      grid_name);
+  m_required_fields.emplace("cflx_k0", scalar2d_layout_col, K,      grid_name);
+  m_required_fields.emplace("wsx",     scalar2d_layout_col, K,      grid_name);
+  m_required_fields.emplace("wsy",     scalar2d_layout_col, K,      grid_name);
+  m_required_fields.emplace("shoc_qv", scalar3d_layout_mid, Qunit,  grid_name);
 
-  FieldLayout scalar3d_layout { {CO,VL}, {ncols,nlevs} }; // Note that C++ and Fortran read array dimensions in reverse
+  m_computed_fields.emplace("t",       scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("shoc_qv", scalar3d_layout_mid, Qunit,  grid_name);
 
-  // Input
-  m_required_fields.emplace("dp", scalar3d_layout,  Pa, grid->name());
+  // Input variables
+  m_required_fields.emplace("host_dx", scalar2d_layout_col, m,  grid_name);
+  m_required_fields.emplace("host_dy", scalar2d_layout_col, m,  grid_name);
+  m_required_fields.emplace("pmid",    scalar3d_layout_mid, Pa, grid_name);
+  m_required_fields.emplace("pint",    scalar3d_layout_int, Pa, grid_name);
+  m_required_fields.emplace("pdel",    scalar3d_layout_mid, Pa, grid_name);
+  m_required_fields.emplace("phis",    scalar2d_layout_col, m,  grid_name);
 
-  // Input-output
+  // Input/Output variables
+  m_required_fields.emplace("s",        scalar3d_layout_mid, J/kg,        grid_name);
+  m_required_fields.emplace("tke",      scalar3d_layout_mid, (m*m)/(s*s), grid_name);
+  m_required_fields.emplace("u",        scalar3d_layout_mid, m/s,         grid_name);
+  m_required_fields.emplace("v",        scalar3d_layout_mid, m/s,         grid_name);
+  m_required_fields.emplace("wthv_sec", scalar3d_layout_mid, K*(m/s),     grid_name);
+  m_required_fields.emplace("tkh",      scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_required_fields.emplace("tk",       scalar3d_layout_mid, (m*m)/s,     grid_name);
+  m_required_fields.emplace("shoc_ql",  scalar3d_layout_mid, Qunit,           grid_name);
+
+  m_computed_fields.emplace("s",        scalar3d_layout_mid, J/kg,        grid_name);
+  m_computed_fields.emplace("tke",      scalar3d_layout_mid, (m*m)/(s*s), grid_name);
+  m_computed_fields.emplace("u",        scalar3d_layout_mid, m/s,         grid_name);
```

Comment:
Are we planning on having two separate fields for the two components of the horiz velocity? I am guessing most uses of horiz velocity will need both components, so a vector velocity might be better (localized mem access).

Reply:
Yes, this is something I plan to do for the (ever growing) shoc optimization PR(s). It will also eliminate the need for a temporary variable in the tridiag solver call..